lib/overcommit/hook/pre_commit/java_checkstyle.rb

```diff
@@ -3,18 +3,25 @@ module Overcommit::Hook::PreCommit
   #
   # @see http://checkstyle.sourceforge.net/
   class JavaCheckstyle < Base
-    MESSAGE_REGEX = /^(\[[^\]]+\]\s+)?(?<file>(?:\w:)?[^:]+):(?<line>\d+)/
+    MESSAGE_REGEX = /^(?<type>\[[^\]]+\]\s+)?(?<file>(?:\w:)?[^:]+):(?<line>\d+)/
+
+    MESSAGE_TYPE_CATEGORIZER = lambda do |type|
+      # Type may be nil if checkstyle doesn't output a 'tag'
+      type ||= ''
+
+      type.include?('WARN') || type.include?('INFO') ? :warning : :error
```

Comment:
Did you want to still report warnings for `INFO` messages, or just ignore them entirely?